### PR TITLE
storage: http(s) backend

### PR DESCRIPTION
**Pull request by adamstruck**
Supports GET requests for public files.

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -30,7 +40,7 @@ type Backend interface {
 	PutFile(ctx context.Context, url string, path string) error
 	// Determines whether this backends supports the given request (url/path/class).
 	// A backend normally uses this to match the url prefix (e.g. "s3://")
-	Supports(url string) error
+	Supports(url string, class tes.FileType, op Operation) error
```
Comment (buchanae) [inline storage/storage.go]: This seems fine, but we could also try supporting a put.

File: storage/storage.go [hunk 2]
```diff
@@ -16,6 +16,16 @@ import (
...
 )
 
+// Operation is a type used to define funnel storage operations
+type Operation int
+
+const (
+	// Get represents the operation Get(ctx context.Context, rawurl string, hostPath string, class tes.FileType) error
+	Get Operation = iota
+	// Put represents the operation Put(ctx context.Context, rawurl string, hostPath string, class tes.FileType) error
+	Put
```
Comment (buchanae) [inline storage/storage.go]: Probably better if these are private

Comment (buchanae) [reply storage/storage.go]: Oh, except, I guess the Backend interface needs to be private too, then.

File: storage/storage.go (hunk 1 above)
Comment (adamstruck) [reply storage/storage.go]: I cant think of a use case where there is going to be a server accepting anonymous POST requests.

File: storage/storage.go (hunk 2 above)
Comment (adamstruck) [reply storage/storage.go]: The worker accesses these in the input and output validation steps. 

The other option is to have a two methods: `SupportsGet` and `SupportsPut`.

File: storage/storage.go (hunk 1 above)
Comment (buchanae) [reply storage/storage.go]: One case would be an http endpoint inside a private network, but it doesn't always need to be anonymous.

**Pull request merged**